libcloud/container/drivers/rancher.py

```diff
@@ -0,0 +1,1212 @@
+import base64
+import shlex
+
+try:
+    import simplejson as json
+except:
+    import json
+
+from libcloud.utils.py3 import httplib
+from libcloud.utils.py3 import b
+
+from libcloud.common.base import JsonResponse, ConnectionUserAndKey
+from libcloud.common.types import InvalidCredsError
+
+from libcloud.container.base import (Container, ContainerDriver,
+                                     ContainerImage)
+
+from libcloud.container.providers import Provider
+from libcloud.container.types import ContainerState
+
+VALID_RESPONSE_CODES = [httplib.OK, httplib.ACCEPTED, httplib.CREATED,
+                        httplib.NO_CONTENT]
+
+
+class RancherResponse(JsonResponse):
+
+    def parse_body(self):
+        if len(self.body) == 0 and not self.parse_zero_length_body:
+            return self.body
+        valid_content_types = ['application/json',
+                               'application/json; charset=utf-8']
+        content_type = self.headers.get('content-type')
+        if content_type in valid_content_types:
+            return json.loads(self.body)
+
+    def parse_error(self):
+        if self.status == 401:
+            raise InvalidCredsError('Invalid credentials')
+        return self.body
```

Comment:
I'm not talking about adding an init to this class, I mean that it should be `return super().parse_error()`.  because looking at the base class, it looks like `parse_error` is suppose to return a parsed response, not just the raw response body.